### PR TITLE
Merge fixes to main

### PR DESCRIPTION
**Pull request by UnsignedByte**

### CODE REVIEW

File: hls/indcpa.cpp [hunk 1]
```diff
@@ -207,7 +213,11 @@ void indcpa_keypair_derand(bit8_t pk[KYBER_INDCPA_PUBLICKEYBYTES],
   bit8_t nonce = 0;
   polyvec a[KYBER_K], e, pkpv, skpv;
 
-  memcpy(buf, coins, KYBER_SYMBYTES);
+  //memset(buf, 0, KYBER_SYMBYTES);
+  for (i = 0; i < KYBER_SYMBYTES; i++) {
+    buf[i] = 0;
+  }
+  // memcpy(buf, coins, KYBER_SYMBYTES);
```
Comment (UnsignedByte) [inline hls/indcpa.cpp:220]: This seems odd to me, the ref originally set this to randombytes, is 0 safe here? @fangzhonglyu

Comment (fangzhonglyu) [inline hls/indcpa.cpp]: This is a temporart thing. Right now this will guarantee deterministic behavior so we can compare with ref

**Comment by UnsignedByte** [comment]
I'm going to change coins to generate random values (with seed) and change ref to match that insteaed

**Comment by UnsignedByte** [comment]
I'm worried that 0 only means we won't be testing properly

**Pull request merged**